DirectEmployers/MyJobs
# [PD-3373] cbo fix edit contact view

## What's this PR do?
The main goal of this ticket was to update the add/delete rule/location logic on the manage contact view. In the process, I corrected a couple of additional issues.

1) Now the categories list on manage contact populates properly.. It didn't populate previously
2) Fixed an issue with ajax calls that prevented the loading spinner from appearing during ajax calls
3) Of course, fixed the add/delete for rules and locations.

## Screenshots (if appropriate)

## Where should the reviewer start?
https://secure.my.jobs/cbo/view/

## How should this be manually tested?
Head to manage contact. Verify that the above issues are corrected.
Bonus.. Check redux to ensure the ajax calls are counting properly.

## Any background context you want to provide?
These are pretty ugly bugs and this PR should be merged before staging push.
